llarp/messages/relay_status.cpp

```diff
@@ -0,0 +1,295 @@
+#include <messages/relay_status.hpp>
+
+#include <crypto/crypto.hpp>
+#include <messages/path_confirm.hpp>
+#include <path/path.hpp>
+#include <router/abstractrouter.hpp>
+#include <util/bencode.hpp>
+#include <util/buffer.hpp>
+#include <util/logger.hpp>
+#include <util/logic.hpp>
+#include <nodedb.hpp>
+
+#include <functional>
+
+namespace llarp
+{
+  struct LRSM_AsyncHandler : public std::enable_shared_from_this< LRSM_AsyncHandler >
+  {
+    using HopHandler_ptr = std::shared_ptr< llarp::path::IHopHandler >;
+
+    std::array< EncryptedFrame, 8 > frames;
+    PathID_t pathid;
+    uint64_t status = 0;
+    HopHandler_ptr path;
+    AbstractRouter *router;
+
+    LRSM_AsyncHandler(const std::array< EncryptedFrame, 8> &_frames,
+                      const PathID_t& _pathid, uint64_t _status,
+                      HopHandler_ptr _path, AbstractRouter *_router)
+        : frames(_frames), pathid(_pathid), status(_status), path(_path), router(_router)
+    {
+    }
+
+    ~LRSM_AsyncHandler()
+    {
+    }
+
+    void handle()
+    {
+      path->HandleLRSM(status, frames, router);
+    }
+
+    void queue_handle()
+    {
+      auto func = std::bind(&llarp::LRSM_AsyncHandler::handle, shared_from_this());
+      llarp_threadpool_queue_job(router->pathContext().Worker(), func);
+    }
+
+  };
+
+
+  LR_StatusMessage::~LR_StatusMessage()
```

Comment:
empty destructors aren't required.

Reply:
copied from LR_CommitMessage, should I change both?

Reply:
addressed